src/ansys/fluent/core/filereader/case_file.py

```diff
@@ -285,29 +292,9 @@ def _get_processed_string(input_string: bytes) -> str:
     return string_identifier + rp_vars_str.split(string_identifier)[1]
 
 
-def _get_case_filepath(project_dir_path: str) -> str:
-    """Gets case file path within the provided project directory path.
-
-    Parameters
-    ----------
-    project_dir_path : str
-        The directory containing the case file
-
-    Returns
-    -------
-    case file path (str)
-    """
-    file_list = list(
-        itertools.chain(
-            *(
-                glob.glob(project_dir_path + r"/**/**-Solve/*.%s" % ext)
-                for ext in ["cas", "cas.h5", "cas.gz"]
-            )
-        )
-    )
-    if len(file_list) < 1:
-        raise RuntimeError(f"No case files are present in: {project_dir_path}")
-    elif len(file_list) > 1:
-        raise RuntimeError(f"More than one case file is present in: {project_dir_path}")
-    else:
-        return file_list[0]
+def _get_case_filepath_from_flprj(flprj_file):
+    parser = etree.XMLParser(recover=True)
+    tree = ET.parse(flprj_file, parser)
+    root = tree.getroot()
+    folder_name = root.find("Metadata").find("CurrentSimulation").get("value")[5:-1]
+    return root.find(folder_name).find("Input").find("Case").find("Target").get("value")
```

Comment:
Actually here you should probably cross-reference via display name. I have saved an .flprj file with whitespace in the study name:
```
<Project type="object" class="PFolder">
...
        <CurrentSimulation class="string" value="LINK:New Study/"/>
...
    </Metadata>
    <New_Study type="object" class="PFolder" displayName="New Study">
...
```
You can look for the "PFolder" objects which has displayName matching the CurrentSimulation value.